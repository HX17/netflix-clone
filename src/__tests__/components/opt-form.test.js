import React from "react";
import { render } from "@testing-library/react";
import OptForm from "../../components/opt-form";

describe("<OptForm />", () => {
  it("renders the <OptForm /> with populated data", () => {
    const { container, getByText, getByPlaceholderText } = render(
      <OptForm>
        <OptForm.Text>
          Ready to watch Netflix? Enter you email to create or restart your
          membership.
        </OptForm.Text>
        <OptForm.Break />
        <OptForm.Input placeholder="Email address" />
        <OptForm.Button>GET STARTED</OptForm.Button>
      </OptForm>
    );

    expect(getByText("GET STARTED")).toBeTruthy();
    expect(
      getByText(
        "Ready to watch Netflix? Enter you email to create or restart your membership."
      )
    ).toBeTruthy();
    expect(getByPlaceholderText("Email address")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
