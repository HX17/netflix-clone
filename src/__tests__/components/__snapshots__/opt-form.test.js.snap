// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptForm /> renders the <OptForm /> with populated data 1`] = `
<div
  class="opt-form__Container-wdq7id-0 bunzPv"
>
  <p
    class="opt-form__Text-wdq7id-1 dxmmPb"
  >
    Ready to watch Netflix? Enter you email to create or restart your membership.
  </p>
  <p
    class="opt-form__Break-wdq7id-4 hDQpoC"
  />
  <input
    class="opt-form__Input-wdq7id-2 iEKa-De"
    placeholder="Email address"
  />
  <button
    class="opt-form__Button-wdq7id-3 bmxngY"
  >
    GET STARTED
     
    <img
      alt="Try now"
      src="/images/icons/chevron-right.png"
    />
  </button>
</div>
`;
